src/components/Target/TargetForm.vue

```diff
@@ -0,0 +1,614 @@
+        inScope: '',
+        outScope: ''
+      }
+      this.target.rootDomains.splice(0)
+      this.rootDomainsTextItems.splice(0)
+      this.validators = {
+        blank: {
+          name: false,
+          repository: false,
+          target: false,
+          command: false
+        },
+        url: {
+          bugBountyUrl: false
+        }
+      }
+    },
+    enableValidationMessageName () {
+      if (this.target.name === '') {
+        this.validators.blank.name = true
+      } else {
+        this.validators.blank.name = false
+      }
+    },
+    enableValidationMessageRootDomains () {
+      if (this.target.rootDomains.length === 0) {
+        this.validators.blank.rootDomains = true
+      } else {
+        this.validators.blank.rootDomains = false
+      }
+    },
+    enableValidationMessageBugBountyUrl () {
+      if (this.target.bugBountyUrl === '') {
+        this.validators.blank.bugBountyUrl = true
+      } else {
+        this.validators.blank.bugBountyUrl = false
+      }
+      if (!this.validateUrl(this.target.bugBountyUrl) && !this.validators.blank.bugBountyUrl) {
+        this.validators.url.bugBountyUrl = true
+      } else {
+        this.validators.url.bugBountyUrl = false
+      }
+    },
+    enableValidationMessageInScope () {
+      if (this.target.inScope === '') {
+        this.validators.blank.inScope = true
+      } else {
+        this.validators.blank.inScope = false
+      }
+    },
+    enableValidationMessageOutScope () {
+      if (this.target.outScope === '') {
+        this.validators.blank.outScope = true
+      } else {
+        this.validators.blank.outScope = false
+      }
+    },
+    enableValidationMessages () {
+      this.enableValidationMessageName()
+      this.enableValidationMessageBugBountyUrl()
+      this.enableValidationMessageRootDomains()
+      this.enableValidationMessageInScope()
+      this.enableValidationMessageOutScope()
+    },
+    setRandomColor () {
+      const predefinedColors = this.$store.state.systemColors
+      this.target.background = predefinedColors[Math.floor(Math.random() * predefinedColors.length)]
+    },
+    verifyPencilStatus () {
+      if (this.isPencilVisibleAndClick) {
+        this.isPencilVisible = true
+      } else {
+        this.isPencilVisible = false
+      }
+    },
+    onEdit () {
+      this.$store.commit('setDetailsLinks', false)
+    },
+    addItemToRootDomains (item) {
+      if (!this.validateUrl(item.value[item.value.length - 1])) {
+        this.rootDomainsTextItems.pop()
+      } else {
+        this.target.rootDomains.push(
+          {
+            root: item.value.slice(-1)[0],
+            id: this.target.rootDomains.length
+          }
+        )
+      }
+    },
+    removeItemToRootDomains (rootDomainParam) {
+      const index = this.target.rootDomains.findIndex(item => item.root === rootDomainParam.value[0])
+      this.target.rootDomains.splice(index, 1)
+    },
+    validateUrl (value) {
```

Comment:
because we are going to use this validation in different place is better move that to a plugin